ResearchKit/Common/ORKFormStepViewController.m

```diff
@@ -891,7 +900,12 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
         ORKTableCellItem *cellItem = section.items[indexPath.row - 1];
         [section.textChoiceCellGroup didSelectCellAtIndexPath:indexPath];
         
+        // formItem's questionType is ORKQuestionTypeSingleChoice even if answerFormat is ORKAnswerFormatBoolean
+        // should check for answerFormat rather than questionType
+        /*
         id answer = (cellItem.formItem.questionType == ORKQuestionTypeBoolean)? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
+        */
+        id answer = ([cellItem.formItem.answerFormat isKindOfClass:[ORKBooleanAnswerFormat class]])? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
```

Comment:
Oh, my bad you are right. `formItem` dose use `impliedAnswerFormat` to
determine questionType.
```

- (ORKQuestionType)questionType {

    return [[self impliedAnswerFormat] questionType];

}
```

On Tue, Nov 17, 2015 at 2:12 PM, Oliver Schäfer <notifications@github.com>
wrote:

> In ResearchKit/Common/ORKFormStepViewController.m
> <https://github.com/ResearchKit/ResearchKit/pull/463#discussion_r45130007>
> :
>
> >          id answer = (cellItem.formItem.questionType == ORKQuestionTypeBoolean)? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
> > +        */
> > +        id answer = ([cellItem.formItem.answerFormat isKindOfClass:[ORKBooleanAnswerFormat class]])? [section.textChoiceCellGroup answerForBoolean] : [section.textChoiceCellGroup answer];
>
> Please note that I'm checking against answerFormat not impliedAnswerFormat
> here by intention. If a form item has a boolean answer format the answer
> should be YES or NO. But the implied answer format will return an array, so
> the answer in fact is [YES] or [NO]. ORKTextChoiceCellGroup has the
> helper method answerForBoolean to convert that array to a bool.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/ResearchKit/ResearchKit/pull/463/files#r45130007>.
>